doc/WinGetConfigure.md

```diff
@@ -0,0 +1,137 @@
+# WinGet and Desired State Configuration
+
+The `winget configure` command is designed to streamline the process of getting your Windows environment to a desired state. WinGet is leveraging PowerShell Core (not Windows PowerShell) and PowerShell DSC (Desired State Configuration) to achieve this goal. WinGet is introducing a new YAML formatted file to define the desired state in a declarative manner rather than an imperative manner. Our goal is to incorporate this new file and it's syntax into PowerShell DSC v3 in the future.
+
+Many users will be new to "Desired State Configuration" as well as the concept of "Configuration as Code". If you want to understand the basics of what is happening when you run `winget configure <configuration file>` you can take a look at our "PowerShell DSC Primer" below.
+
+## PowerShell DSC Primer
+
+Getting Started with PowerShell DSC in the context of learning the basics of `winget configure`.
+
+This section outlines the process of ensuring you have a PowerShell DSC v2 compatible version of PowerShell. Then it walks you through the process of installing PowerShell Desired State Configuration module. Next it walks you through installing the PowerShell Desired State Resources module. Finally you are given examples to run `Invoke-DscResource` and validate the behavior.
+
+### Ensure you are using PowerShell 7.2 or later
+
+First you need to have PowerShell 7.2 or later. You can run `$PSVersionTable` in PowerShell if you see the following result, you need to switch to a more recent version of PowerShell or install one.
+
+```PowerShell
+PS C:\Users\denelon> $PSVersionTable
+
+Name                           Value
+----                           -----
+PSVersion                      5.1.22621.963
+PSEdition                      Desktop
+PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
+BuildVersion                   10.0.22621.963
+CLRVersion                     4.0.30319.42000
+WSManStackVersion              3.0
+PSRemotingProtocolVersion      2.3
+SerializationVersion           1.1.0.1
+```
```

Comment:
Do we need the whole version table, or would `$PSVersionTable.PSVersion.ToString()` be enough?

Reply:
I'm assuming no familiarity with PowerShell or DSC in this document. So I'm "trying" to keep things a bit simple so I don't necessarily need to explain all the syntax. I'm open to suggestions.